codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Headers.java

```diff
@@ -354,4 +286,61 @@ public ByteString path() {
     public ByteString status() {
         return get(PseudoHeaderName.STATUS.value());
     }
+
+    @Override
+    public int hashCode() {
+        return size();
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (!(other instanceof DefaultHttp2Headers)) {
+            return false;
+        }
+        DefaultHttp2Headers headers = (DefaultHttp2Headers) other;
+        if (size() != headers.size()) {
+            return false;
+        }
+        Comparator<ByteString> valueComparator = ByteString.DEFAULT_COMPARATOR;
+        for (ByteString  name : names()) {
+            List<ByteString> otherValues = headers.getAll(name);
+            List<ByteString> values = getAll(name);
```

Comment:
isn't there some better way to do this ? We create and full two new ArrayLists for it.